### DISCUSSION TITLE
Can I use useSWR for all data fetching in my app?

### DISCUSSION THREAD

**Discussion by HarshBoricha404** [vercel/swr#1894] [Q&A]
Can someone please explain if I could use useSWR everywhere in my app. Would that be okay to do so, or it may impact performance since swr will cache everything.

**Comment by icyJoseph** [comment]
React-Query and SWR are more like data management tools, to get data from an external source, you'll need to implement a fetcher, usually built with fetch/axios.

You can make custom hooks that expose the data to avoid having to pass the fetcher function around.

**Reply by HarshBoricha404** [reply]
Thanks for the insight. My concern is would that be okay to fetch everything with useSWR?
